pkg/client/client.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 Google LLC.
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"context"
+	"encoding/json"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/kubectl/pkg/scheme"
+	"k8s.io/kubectl/pkg/util"
+	"sigs.k8s.io/cli-utils/pkg/object"
+)
+
+// client is the client to update object in the API server.
+type client struct {
+	client     dynamic.Interface
+	restMapper meta.RESTMapper
+}
+
+func NewClient(d dynamic.Interface, mapper meta.RESTMapper) *client {
+	return &client{
+		client:     d,
+		restMapper: mapper,
+	}
+}
+
+// Update updates an object using dynamic client
+func (uc *client) Update(ctx context.Context, meta object.ObjMetadata, obj *unstructured.Unstructured, options *metav1.UpdateOptions) error {
+	r, err := uc.resourceInterface(meta)
+	if err != nil {
+		return err
+	}
+	if options == nil {
+		options = &metav1.UpdateOptions{}
+	}
+	_, err = r.Update(ctx, obj, *options)
```

Comment:
Does this result in an `UPDATE` or a `PATCH` call to the apiserver? Just want to make sure we do this the same way as a regular apply operation.

Reply:
It's an `UPDATE` call. I didn't use the `PATCH` call since we should only change the `metadata.annotation`.  Only two annotations are updated: the `owning-inventory` and the last-applied-config annotation. If we compute them correctly, we don't need to use the `PATCH` call.

Reply:
For more context: the `Update` call translates to a REST `Put` with the entire object in the body.